packages/flutter_tools/lib/src/build_system/targets/windows.dart

```diff
@@ -17,46 +37,99 @@ class UnpackWindows extends Target {
   @override
   List<Source> get inputs => const <Source>[
     Source.pattern('{FLUTTER_ROOT}/packages/flutter_tools/lib/src/build_system/targets/windows.dart'),
-    Source.artifact(Artifact.windowsDesktopPath, mode: BuildMode.debug),
+  ];
+
+  @override
+  List<Source> get outputs => const <Source>[];
+
+  @override
+  List<String> get depfiles => <String>['windows_engine_sources.d'];
+
+  @override
+  List<Target> get dependencies => const <Target>[];
+
+  @override
+  Future<void> build(Environment environment) async {
+    final String artifactPath = environment.artifacts.getArtifactPath(Artifact.windowsDesktopPath);
+    final String outputPrefix = environment.fileSystem.path.join(
+      environment.projectDir.path,
+      'windows',
+      'flutter',
+      'ephemeral',
+    );
+    final Depfile depfile = unpackDesktopArtifacts(
+      fileSystem: environment.fileSystem,
+      artifacts: _kWindowsArtifacts,
+      artifactPath: artifactPath,
+      outputPrefix: outputPrefix,
+    );
+    final DepfileService depfileService = DepfileService(
+      fileSystem: environment.fileSystem,
+      logger: environment.logger,
+    );
+    depfileService.writeToFile(
+      depfile,
+      environment.buildDir.childFile('windows_engine_sources.d'),
```

Comment:
This string appears twice; would be nice to have a constant instead.

Reply:
Done